### ISSUE TITLE
bug: Query params not forwarded when using HTTP Proxy with API Gateway REST API

### ISSUE THREAD

**Issue by ajhaining** [localstack/localstack#5125]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I am trying to build an API Gateway HTTP Proxy fallback for a REST API. I have the following API Gateway resources configured locally with localstack.

```json
{
  "httpMethod": "ANY",
  "authorizationType": "NONE",
  "apiKeyRequired": false,
  "requestParameters": {
    "method.request.path.proxy": true
  },
  "methodIntegration": {
    "type": "HTTP_PROXY",
    "httpMethod": "ANY",
    "uri": "https://httpbin.org/anything/{proxy}",
    "requestParameters": {
      "integration.request.path.proxy": "method.request.path.proxy"
    },
    "passthroughBehavior": "WHEN_NO_MATCH",
    "timeoutInMillis": 29000,
    "cacheNamespace": "9b6e854f",
    "cacheKeyParameters": []
  }
}

```

When I call this endpoint with `GET http://localhost:4566/restapis/vt7zvfoavm/local/_user_request_/foo/bar/baz?myParam=12345678` I receive the following response (Using [HTTPbin](https://httpbin.org/) here to return the request as the response)

```jsonc
{
    "args": {}, // no query params in args
    "data": "",
    "files": {},
    "form": {},
    "headers": {
        "--Apigw-Request-Region--": "eu-west-1",
        "Accept": "*/*",
        "Accept-Encoding": "gzip, deflate, br",
        "Authorization": "AWS4-HMAC-SHA256 Credential=__internal_call__/20160623/us-east-1/apigateway/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-target, Signature=1234",
        "Host": "localhost",
        "Postman-Token": "134b9852-7589-4a54-b730-5444588ae024",
        "Remote-Addr": "172.17.0.1",
        "User-Agent": "PostmanRuntime/7.28.4",
        "X-Amzn-Trace-Id": "Root=1-61b9bc31-6c2834ad38c63f456284cdd9",
        "X-Localstack-Edge": "http://localhost:4566",
        "X-Localstack-Tgt-Api": "apigateway"
    },
    "json": null,
    "method": "GET",
    "origin": "172.17.0.1, localhost:4566, 127.0.0.1, localhost:4566, 92.27.80.225",
    "url": "https://localhost/anything/foo/bar/baz" // no query params on url
}
```

The query parameter `myParam=12345678` is stripped from the request, despite being a valid part of the `{proxy}` path.

I used the same configuration to create resources in AWS:

```json
{
  "httpMethod": "ANY",
  "authorizationType": "NONE",
  "apiKeyRequired": false,
  "requestParameters": {
    "method.request.path.proxy": true
  },
  "methodIntegration": {
    "type": "HTTP_PROXY",
    "httpMethod": "ANY",
    "uri": "https://httpbin.org/anything/{proxy}",
    "connectionType": "INTERNET",
    "requestParameters": {
      "integration.request.path.proxy": "method.request.path.proxy"
    },
    "passthroughBehavior": "WHEN_NO_MATCH",
    "timeoutInMillis": 29000,
    "cacheNamespace": "445f0a",
    "cacheKeyParameters": []
  }
}
```

And the response is as expected:

```jsonc
// GET https://redacted.execute-api.eu-west-1.amazonaws.com/v1/foo/bar/baz?myParam=12345678
{
    "args": {
        "myParam": "12345678" // query param passed through
    },
    "data": "",
    "files": {},
    "form": {},
    "headers": {
        "Accept": "*/*",
        "Accept-Encoding": "gzip, deflate, br",
        "Cloudfront-Forwarded-Proto": "https",
        "Cloudfront-Is-Desktop-Viewer": "true",
        "Cloudfront-Is-Mobile-Viewer": "false",
        "Cloudfront-Is-Smarttv-Viewer": "false",
        "Cloudfront-Is-Tablet-Viewer": "false",
        "Cloudfront-Viewer-Country": "GB",
        "Host": "httpbin.org",
        "Postman-Token": "41397d0f-d04b-4b39-8110-5a5d9684e86d",
        "User-Agent": "PostmanRuntime/7.28.4",
        "X-Amz-Cf-Id": "Sb6KMrHBuK0V1zyn9TdmKm0HreYhZrOmgHHec0mPD5jR0YjGpzDDEg==",
        "X-Amzn-Apigateway-Api-Id": "redacted",
        "X-Amzn-Trace-Id": "Self=1-61b9be60-5a0d8f4d09b8f5c554bbe723;Root=1-61b9be5f-37e6655166d36d2a4b330d4f"
    },
    "json": null,
    "method": "GET",
    "origin": "92.27.80.225, 130.176.222.55, 108.128.160.192",
    "url": "https://httpbin.org/anything/foo/bar/baz?myParam=12345678" // query param on url
}
```

I have also tried explicitly mapping query parameters with `"integration.request.querystring.myParam": "method.request.querystring.myParam"`, this doesn't work and also doubles up the query param in the AWS environment.

### Expected Behavior

I expect query parameters to be passed through on HTTP_PROXY requests as they are in the AWS equivalent.

### How are you starting LocalStack?

With the `localstack` script

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

I'm using `localstack start`

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

I'm using Terraform to configure the environments:

```hcl
resource "aws_api_gateway_resource" "test-api-proxy-resource" {
  rest_api_id = aws_api_gateway_rest_api.test-api.id
  parent_id   = aws_api_gateway_rest_api.test-api.root_resource_id
  path_part   = "{proxy+}"
}

resource "aws_api_gateway_method" "test-api-proxy-method" {
  rest_api_id   = aws_api_gateway_rest_api.test-api.id
  resource_id   = aws_api_gateway_resource.test-api-proxy-resource.id
  http_method   = "ANY"
  authorization = "NONE"

  request_parameters = {
    "method.request.path.proxy" = true
  }
}

resource "aws_api_gateway_integration" "test-api-proxy-method-integration" {
  rest_api_id             = aws_api_gateway_rest_api.test-api.id
  resource_id             = aws_api_gateway_resource.test-api-proxy-resource.id
  http_method             = aws_api_gateway_method.test-api-proxy-method.http_method
  type                    = "HTTP_PROXY"
  integration_http_method = "ANY"
  uri                     = "https://httpbin.org/anything/{proxy}"
  passthrough_behavior    = "WHEN_NO_MATCH"
  request_parameters = {
    "integration.request.path.proxy" = "method.request.path.proxy"
  }
}
```

I use `aws{local} apigateway get-method --rest-api-id <ID> --resource-id <ID> --http-method ANY` to get the configuration out of the environment.

### Environment

```markdown
- OS: macOS 12.0.1
- LocalStack: 0.13.1 and 0.13.0.11
```

**Comment by calvernaz** [comment]
Thanks for reporting @ajhaining. This should be fixed in #5212, can you please pull the latest Docker image and give it another try?

**Comment by ajhaining** [comment]
> Thanks for reporting @ajhaining. This should be fixed in #5212, can you please pull the latest Docker image and give it another try?

This is great, thanks @calvernaz, I'll be working with it this week, I'll let you know how it goes. Thanks again.
